provider/src/cache.ts

```diff
@@ -151,3 +55,151 @@ export const storeContractStatusInKV = async (
     console.error('Error storing contract status in KV:', error)
   }
 }
+
+/**
+ * Get a cached response from KV storage
+ * @param request The RPC request
+ * @param chainId The chain ID
+ * @param env The environment variables
+ * @returns The cached response or null if not found
+ *
+ * First attempts to retrieve and parse as JSON directly. If that fails,
+ * falls back to retrieving as text and parsing manually. If the text
+ * parsing also fails, deletes the corrupted cache entry.
+ */
+export const getResponseFromKV = async (
+  request: RpcRequest,
+  chainId: string,
+  env: Env
+): Promise<any> => {
+  if (!env.REQUEST_CACHE) {
+    return null
+  }
+
+  const cacheKey = generateRequestCacheKey(request)
+
+  try {
+    // Try to get and parse as JSON
+    const cached = await env.REQUEST_CACHE.get(cacheKey, 'json')
+    if (cached) {
+      return cached
+    }
+  } catch (error) {
+    // If JSON parsing fails, try simple text retrieval and parse it
+    try {
+      const rawText = await env.REQUEST_CACHE.get(cacheKey, 'text')
+      if (rawText) {
```

Comment:
This is bad. We should decide to either store json OR store text and not mix things.